idb/migration/migration_test.go

```diff
@@ -329,3 +330,105 @@ func TestSuccessfulMigration(t *testing.T) {
 		})
 	}
 }
+
+// TestAvailabilityChannelCloses tests that the migration object closes the availability
+// channel when blocking migrations finish. If it is not the case, this test will deadlock
+// (and timeout).
+func TestAvailabilityChannelCloses(t *testing.T) {
+	// Migration 2 reads on this channel.
+	migrationTwoChannel := make(chan struct{})
+
+	tasks := []Task{
+		{
+			MigrationID: 1,
+			Handler: func() error {
+				return nil
+			},
+			DBUnavailable: true,
+		},
+		{
+			MigrationID: 2,
+			Handler: func() error {
+				<-migrationTwoChannel
+				return nil
+			},
+		},
+	}
+
+	m, err := MakeMigration(tasks, nil)
+	require.NoError(t, err)
+
+	availableCh := m.RunMigrations()
+	_, ok := <-availableCh
+	assert.False(t, ok)
```

Comment:
This is a test to see if the `availableCh` has been closed? I thought that this was not recommended. You could do something like this instead:
```
	select {
	case <-availableCh:
		assert.Fail("The channel should not have been closed.")
	default:
		
	}
```

Reply:
No, I can't do this. Due to non-determinism, I can't be sure if the channel is already closed or not.

Reply:
How would it close? The migration should be blocked by `migrationTwoChannel`

Reply:
Only the first migration is blocking.

Reply:
I think you need to avoid the `_, ok := <-ch` pattern, what are you expecting it to do?

If you want to ensure the channel is closed use:
```
for _ := range ch {
}
```

Reply:
Your code snippet doesn't verify that nothing get written on the channel, so it's a weaker check.